features/nvstore/source/nvstore.h

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NVSTORE_H
+#define __NVSTORE_H
+
+// These addresses need to be configured according to board (in mbed_lib.json)
+#if defined(NVSTORE_AREA_1_ADDRESS) && defined(NVSTORE_AREA_1_SIZE) && \
+    defined(NVSTORE_AREA_2_ADDRESS) && defined(NVSTORE_AREA_2_SIZE) && \
+    defined(DEVICE_FLASH)
+#define NVSTORE_ENABLED 1
+#endif
+
+#if NVSTORE_ENABLED
+#include <stdint.h>
+#include <stdio.h>
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+
+enum {
+    NVSTORE_SUCCESS                =  0,
+    NVSTORE_READ_ERROR             = -1,
+    NVSTORE_WRITE_ERROR            = -2,
+    NVSTORE_NOT_FOUND              = -3,
+    NVSTORE_DATA_CORRUPT           = -4,
+    NVSTORE_BAD_VALUE              = -5,
+    NVSTORE_BUFF_TOO_SMALL         = -6,
+    NVSTORE_FLASH_AREA_TOO_SMALL   = -7,
+    NVSTORE_OS_ERROR               = -8,
+    NVSTORE_BUFF_NOT_ALIGNED       = -9,
+    NVSTORE_ALREADY_EXISTS         = -10,
+};
+
+#ifndef NVSTORE_MAX_KEYS
+#define NVSTORE_MAX_KEYS 16
+#endif
+
+class NVStore {
+public:
+
+/**
+ * @brief As a singleton, return the single instance of the class.
+ *        Reason for this class being a singleton is the following:
+ *        - Ease the use for users of this class not having to coordinate instantiations.
+ *        - Lazy instantiation of internal data (which we can't achieve with simple static classes).
+ *
+ * @returns Singleton instance reference.
+ */
+    static NVStore& get_instance()
+    {
+        // Use this implementation of singleton (Meyer's) rather than the one that allocates
+        // the instance on the heap, as it ensures destruction at program end (preventing warnings
+        // from memory checking tools such as valgrind).
+        static NVStore instance;
+        return instance;
+    }
+
+    virtual ~NVStore();
+
+/**
+ * @brief Returns number of keys.
+ *
+ * @returns Number of keys.
+ */
+    uint16_t get_max_keys();
```

Comment:
can be `const`, similar other methods like `size()`

Reply:
This one can indeed be const and will be modified. The size function can't, as it lazily calls the init function, required in order to determine the NVStore size).